Automatic list with overviews of inlcuded area definitions for the documentation

BENR0 opened the pull request:
Based on https://github.com/pytroll/pyresample/pull/450 this generates overviews for all `AreaDefinitions` included with Satpy and adds them to the resampling section.

I noticed that a quite large number of `AreaDefinitions` are included which make the section really long now.
Some areas are not included because only `AreaDefinitions` currently have `_repr_html_` (see PR above).

 - [x] Closes #1206  <!-- remove if there is no corresponding issue, which should only be the case for minor changes -->
 - [x] Fully documented <!-- remove if this change should not be visible to users, e.g., if it is an internal clean-up, or if this is part of a larger project that will be documented later -->

gerritholl commented:
This looks interesting.  How easy would it be for users to use this feature to generate overviews of areas from their own local YAML files?

mraspaud commented on doc/source/area_def_list.py:
```diff
@@ -0,0 +1,51 @@
+from satpy.resample import get_area_def, get_area_file
+
+
+def generate_area_def_list():
+    """Create list of available area definitions with overview plot.
+
+    Returns:
+        str
+    """
+    area_list = []
+
+    template = ("{area_name}\n"
+                "{n:->{header_title_length}}\n\n"
+                ".. raw:: html\n\n"
+                "{content}\n\n"
+                "     <hr>\n\n")
+
+    area_file = get_area_file()[0]
+    for aname in list(_read_yaml_area_file_content(area_file).keys()):
+        area = get_area_def(aname)
+        if hasattr(area, "_repr_html_"):
+            content = "\n".join([x.rjust(len(x) + 5) for x in area_repr(area, include_header=False).split("\n")])
+            area_list.append(template.format(area_name=aname, n="", header_title_length=len(aname),
+                                             content=content))
+        else:
+            pass
```
Doesn't this function rather belong to pyresample?

BENR0 replied:
I think it doesn't really make sense to include in in pyresample because there is no area definition yaml included and this is only useful to generate a rst with a list of area definitions for the documentation.

mraspaud replied:
It is true that pyresample does not have a default yaml file, but it is pyresample that contains all the functions for reading such yaml files. And also, I would argue that for the sake of separation of concerns, it should be included there.
In practice, I wouldn't be against having a default yaml file with a few generic areas in pyresample by the way, that would probably help potential pyresample users in the future...

gerritholl replied:
Wouldn't `areas.yaml` fit better with pyresample entirely?  But pyresample does not have a concept of `SATPY_CONFIG_PATH` or a config directory search, so it wouldn't know where to find `areas.yaml` anyway?

djhoese replied:
Not that anyone asked for my opinion, but I agree with wanting this in pyresample. I'm not a big fan of a default set of areas in pyresample except only as a few examples (like global lon/lat, one LCC, one polar-stereographic, etc)...generic areas. Pyresample is the low-level functionality, but it doesn't know anything about satellite-specific areas or that it is used for "science" at all so to me it is a fine line.

mraspaud replied:
A while ago, I found a small list of generic areas, like europe, north and south america, etc. I'll try to find it again, so we can use that.

pnuu replied:
As far as I see, those are just area _names_ without defining anything `AreaDefinition`y that we could use.

BENR0 commented:
@gerritholl Curently not that easy I guess. But could easily be refactored to be available to users I think.

mraspaud commented:
> @gerritholl Curently not that easy I guess. But could easily be refactored to be available to users I think.

Especially if it was in pyresample ;)

BENR0 commented:
@mraspaud @gerritholl ok so where should this go in pyresample? Should I include it in https://github.com/pytroll/pyresample/pull/450 or make a separate PR?

pnuu reviewed (approved):
LGTM. A screenshot would be nice :-)

pnuu commented:
There are some problems with building the documentation, but they seem to be because requiring https://github.com/pytroll/pyresample/pull/450 to be merged and released first.

BENR0 commented:
Since there are quite a few `AreaDefinition`s in the yaml file I added a searchable table similar to the reader table with links to the definitions so the user does not have to scroll through the whole document. Currently this is still a bit messy since I just included the generation in the `conf.py` of the documentation for demonstration purposes. If this is wanted I will refactor this. It could be integrated into the generation function I already moved to pyresample but there on its own it would just generated a static table because the searching and sorting depends on the datatable css and js which is integrated in the satpy documentation.

Below you can see how it would look like. One think to note is that all the area definitions also appear in the toc tree which is not a huge deal but not that nice I guess. I think that has something to do with the toc maxdepth setting but I couldn't get that fixed somehow. So if somebody has an idea about that please share.

Apart from that I notice that there are two area definitions I think belong to `DynamicAreaDefinition`s?:
- omerc_bb
- laea_bb
Right now these produce broken links in the table because obviously there can be no overview generated for those.

![area_def_table](https://user-images.githubusercontent.com/12115839/205655219-0976b700-c09f-4b78-b999-1b1e3667f714.png)

mraspaud commented:
Just as a reminder for whoever is looking for a status update, this is waiting for completion of https://github.com/pytroll/pyresample/pull/450

mraspaud commented:
https://github.com/pytroll/pyresample/pull/450 is now merged. Do we need a pyresample release before this can work?

BENR0 commented:
I think so because if I see correctly read the docs uses a conda installed pyresample to build the docs.
While the table should build fine in either case it would be nice to have the area names in the yaml file fixed (even if there is no agreement on a file naming pattern yet) #2310.

gerritholl commented:
You can add yourself to `AUTHORS.md` while you're at it :)
